### PR TITLE
Move `s390x-unknown-linux-musl` to build.sh

### LINKED ISSUE #2085: Move `s390x-unknown-linux-musl` to the build check
It's a tier 3 target and it should be in build.sh.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #2085
r? @ghost

### CODE REVIEW

**Comment by JohnTitor** [comment]
Failure:
```
error[E0412]: cannot find type `ipc_perm` in the crate root
  --> src/unix/linux_like/linux/musl/b64/mod.rs:51:25
   |
51 |         pub shm_perm: ::ipc_perm,
   |                         ^^^^^^^^ not found in the crate root

error[E0412]: cannot find type `ipc_perm` in the crate root
  --> src/unix/linux_like/linux/musl/b64/mod.rs:64:25
   |
64 |         pub msg_perm: ::ipc_perm,
   |                         ^^^^^^^^ not found in the crate root

error[E0412]: cannot find type `__rlimit_resource_t` in the crate root
   --> src/unix/linux_like/linux/musl/b64/s390x.rs:205:25
    |
205 | pub const RLIMIT_RSS: ::__rlimit_resource_t = 5;
    |                         ^^^^^^^^^^^^^^^^^^^ not found in the crate root

error[E0412]: cannot find type `__rlimit_resource_t` in the crate root
   --> src/unix/linux_like/linux/musl/b64/s390x.rs:206:24
    |
206 | pub const RLIMIT_AS: ::__rlimit_resource_t = 9;
    |                        ^^^^^^^^^^^^^^^^^^^ not found in the crate root

error[E0412]: cannot find type `__rlimit_resource_t` in the crate root
   --> src/unix/linux_like/linux/musl/b64/s390x.rs:207:29
    |
207 | pub const RLIMIT_MEMLOCK: ::__rlimit_resource_t = 8;
    |                             ^^^^^^^^^^^^^^^^^^^ not found in the crate root

error[E0412]: cannot find type `__rlimit_resource_t` in the crate root
   --> src/unix/linux_like/linux/musl/b64/s390x.rs:208:28
    |
208 | pub const RLIMIT_NOFILE: ::__rlimit_resource_t = 7;
    |                            ^^^^^^^^^^^^^^^^^^^ not found in the crate root

error[E0412]: cannot find type `__rlimit_resource_t` in the crate root
   --> src/unix/linux_like/linux/musl/b64/s390x.rs:209:27
    |
209 | pub const RLIMIT_NPROC: ::__rlimit_resource_t = 6;
    |                           ^^^^^^^^^^^^^^^^^^^ not found in the crate root

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0412`.
error: could not compile `libc`
```

File: src/unix/linux_like/linux/musl/mod.rs [hunk 1]
```diff
@@ -644,6 +641,20 @@ pub const PRIO_PROCESS: ::c_int = 0;
 pub const PRIO_PGRP: ::c_int = 1;
 pub const PRIO_USER: ::c_int = 2;
 
+cfg_if! {
+    if #[cfg(target_arch = "s390x")] {
+        pub const POSIX_FADV_DONTNEED: ::c_int = 6;
+        pub const POSIX_FADV_NOREUSE: ::c_int = 7;
+
+        pub const PTRACE_DETACH: ::c_uint = 17;
```
Comment (kaniini) [inline src/unix/linux_like/linux/musl/mod.rs:649]: `PTRACE_DETACH` is defined to the same value regardless of branch.

Comment (JohnTitor) [reply src/unix/linux_like/linux/musl/mod.rs:649]: Note that it's `c_uint` as you defined previously: https://github.com/rust-lang/libc/blob/5fd6540db9b14fae02ff8a11a6102ee99965ae30/src/unix/linux_like/linux/musl/b64/s390x.rs#L358

Comment (kaniini) [reply src/unix/linux_like/linux/musl/mod.rs:649]: I think this should actually be `::c_int` to be aligned with the `ptrace()` definition.  I used a simple awk script to generate the definitions based on the musl headers.

**Pull request merged**